OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -477,8 +481,10 @@ bool CanTargetLocation(Actor self, CPos location, TargetModifiers modifiers, ref
 
 				IsQueued = modifiers.HasModifier(TargetModifiers.ForceQueue);
 
-				// Targeting the terrain is only possible with force-attack modifier
-				if (modifiers.HasModifier(TargetModifiers.ForceMove) || !modifiers.HasModifier(TargetModifiers.ForceAttack))
+				// Targeting the terrain is only possible with force-attack modifier or when TargetsTerrain is set
+				if (!(ab.Info.TargetsTerrain ||
+					modifiers.HasModifier(TargetModifiers.ForceAttack)) ||
+					modifiers.HasModifier(TargetModifiers.ForceMove))
```

Comment:
```suggestion
				// Targeting the terrain is only possible with force-attack modifier.
				if (modifiers.HasModifier(TargetModifiers.ForceMove)
					|| (!ab.Info.TargetsTerrain && !modifiers.HasModifier(TargetModifiers.ForceAttack)))
```

I was more thinking like this